### PR TITLE
[OP-73] Add tests to RTS code

### PR DESCRIPTION
**Pull request by peendebak**
* add tests for RTS
* remove plunger argument to RTS
* fix corner case in RTS
* Cleanup code

@CumulonimbusCalvus @mjwoudstra

### CODE REVIEW

File: src/qtt/algorithms/random_telegraph_signal.py [hunk 1]
```diff
@@ -114,6 +114,19 @@ def plot_two_level_threshold(results, fig=100):
     plt.title('Result of two level threshold processing')
 
 
+def _create_integer_histogram(durations, verbose=0):
+    """ Calculate number of bins, bin edges and histogram for durations
+
+    This method works if the data is sampled at integer durations.
```
Comment (QFer) [inline src/qtt/algorithms/random_telegraph_signal.py]: When durations is empty array then error. When one duration is zero you get a negative bin_edge (-0.5). Is that oke?

File: src/tests/unittests/algorithms/test_random_telegraph_signal.py [hunk 2]
```diff
@@ -49,4 +50,25 @@ def test_RTS(self, fig=None, verbose=0):
...
+                                                                    fig=None, verbose=0)
+        self.assertAlmostEqual(tunnelrate_dn, 1000, -2)
+        self.assertAlmostEqual(tunnelrate_up, 1000, -2)
+        self.assertAlmostEqual(rts_results['fraction_down'], 0.7047, 1)
+        self.assertEqual(rts_results['fraction_down'], 1 - rts_results['fraction_up'])
+
+    def test_create_integer_histogram(self):
+        input_data = np.array([2, 3, 4])
+        counts, bins, bin_size = _create_integer_histogram(input_data)
+        self.assertEqual(input_data.size, np.sum(counts))
```
Comment (QFer) [inline src/tests/unittests/algorithms/test_random_telegraph_signal.py]: Maybe check the expected bin_edges.

File: src/qtt/algorithms/random_telegraph_signal.py [hunk 3]
```diff
@@ -151,26 +163,17 @@ def tunnelrates_RTS(data, samplerate=None, min_sep=2.0, max_sep=7.0, min_duratio
                 tunnelrate_up (float): tunnel rate up in Hz
 
     """
-    if plungers is None:
-        plungers = []
+    if plungers is not None:
+        raise Exception('parameter plungers is not used any more')
```
[the remark this answers is not in the document]
Comment (peendebak) [reply src/qtt/algorithms/random_telegraph_signal.py]: It is not backwards compatible, the `plunger`functionality should not be in the method.
With the exception someone trying to use the old functionality will get an error message describing this, if I remove the argument there might not be an error message at all (if someone does not use keyword arguments but direct arguments).

File: src/qtt/algorithms/random_telegraph_signal.py [hunk 4]
```diff
@@ -114,6 +114,19 @@ def plot_two_level_threshold(results, fig=100):
...
+def _create_integer_histogram(durations, verbose=0):
+    """ Calculate number of bins, bin edges and histogram for durations
+
+    This method works if the data is sampled at integer durations.
+    """
+    numbins = int(np.sqrt(len(durations)))
+    bin_size = int(np.ceil((durations.max() - (durations.min() - .5)) / numbins))
+    # choose bins carefully, since our data is sampled only at discrete times
+    bins = np.arange(durations.min() - .5, durations.max() + bin_size, bin_size)
+    counts, bins = np.histogram(durations, bins=bins)
```
[the remark this answers is not in the document]
Comment (peendebak) [reply src/qtt/algorithms/random_telegraph_signal.py]: `bin_edges` is indeed better

File: src/qtt/algorithms/random_telegraph_signal.py [hunk 5]
```diff
@@ -151,26 +163,17 @@ def tunnelrates_RTS(data, samplerate=None, min_sep=2.0, max_sep=7.0, min_duratio
...
 
-        data = np.array(data.measured)
-    else:
-        plungervalue = []
-        if samplerate is None:
-            raise Exception('samplerate is None')
+        data = np.array(data.default_parameter_array())
+
+    if samplerate is None:
+        raise Exception('samplerate is None')
```
[the remark this answers is not in the document]
Comment (peendebak) [reply src/qtt/algorithms/random_telegraph_signal.py]: Changed the exception and also removed the default value for the argument of the method.

File: src/qtt/algorithms/random_telegraph_signal.py [hunk 6]
```diff
@@ -114,6 +114,23 @@ def plot_two_level_threshold(results, fig=100):
...
 
+def _create_integer_histogram(durations):
+    """ Calculate number of bins, bin edges and histogram for durations
+
+    This method works if the data is sampled at integer durations.
+    """
+    numbins = int(np.sqrt(len(durations)))
+
+    if numbins == 0:
+        raise Exception('cannot create histogram with zero bins')
```
Comment (QFer) [inline src/qtt/algorithms/random_telegraph_signal.py]: Try to avoid Low level Exception class. ValueError is better
